plugin/trino-hive/src/main/java/io/trino/plugin/hive/orc/OrcTypeTranslator.java

```diff
@@ -73,6 +75,9 @@ private OrcTypeTranslator() {}
             return Optional.of(new DateToVarcharCoercer(varcharType));
         }
         if (isVarcharType(fromOrcType)) {
```

Comment:
The scaffolding with `isOrcFile` is done for ORC only.
`OrcPageSourceFactory` is using though `OrcTypeTranslator` (for both partitioned and unpartitioned tables) where we know that we're dealing with ORC.
The `CoercionUtils` class (which makes use of `CoercionContext`) has actually nothing to do with ORC changes at the moment. 

Instead of adding special logic for ORC in the `XToYCoercer` classes, I'd be in favor of having separate coercer classes:

- `XToYCoercer`
- `XToYOrcCoercer`

This change would keep the `XToYCoercer` classes easier to follow.
I don't feel strongly about this chnage because the coercer classes are simple at the moment.

Reply:
> The scaffolding with isOrcFile is done for ORC only.

True 

> OrcPageSourceFactory is using though OrcTypeTranslator (for both partitioned and unpartitioned tables) where we know that we're dealing with ORC.

This one is partially true - Though OrcTypeTranslator is well equipped with handling the coercions for partitioned table as well - We still rely on `CoercionPolicy` or `CoercionUtil` to control the coercion for partitioned layer - Today the coercion are applied at a layer above the file format specfic PageSourceProvider and we don't want to split it for each file format. So still for ORC files in case of partitioned tables we rely on `CoercionUtils` 


>  I'd be in favor of having separate coercer classes:

We can handle them but for boolean, double or float the changes are fairly simple so we have constrained them within a class but for numeric coercions https://github.com/trinodb/trino/pull/20766 we have implemented something like a seperate coercer for ORC